uaclient/config.py

```diff
@@ -698,3 +709,38 @@ def new_f():
         return new_f
 
     return wrapper
+
+
+def redact_config_data(
+    user_config: Optional[state_files.UserConfigData],
+) -> Optional[state_files.UserConfigData]:
+    """Redact the user-config.json file and return its contents."""
+    if user_config:
+        redacted_data = user_config
+        for field in PROXY_FIELDS:
+            value = getattr(redacted_data, field)
+            if value:
+                setattr(
+                    redacted_data, field, util.redact_sensitive_proxy(value)
+                )
+        return redacted_data
+    return None
+
+
+def read_user_config() -> Optional[state_files.UserConfigData]:
+    """Read the user-config.json file and return its contents."""
+    try:
+        if util.we_are_currently_root():
+            return state_files.user_config_file_private.read()
+        public_config = state_files.user_config_file_public.read()
+        return redact_config_data(public_config)
+    except Exception as e:
+        LOG.warning("Error loading user config", exc_info=e)
+        LOG.warning("Using default config values")
+        return None
+
+
+def write_user_config(user_config: state_files.UserConfigData) -> None:
+    """Write the user-config.json file with the given contents."""
+    state_files.user_config_file_public.write(user_config)
+    state_files.user_config_file_private.write(user_config)
```

Comment:
I think we might use a similar structure to what we have for the `machine-token` file:
https://github.com/canonical/ubuntu-pro-client/blob/main/uaclient/files/files.py#L74

You can see that this class is responsible for handling both the public and private version of the machine token file. This would simplify the config module, as it wouldn't need to know about the existence of the separate files

Reply:
@lucasmoura should be good for another look